docs/training/apis-howto/api-core-mnist-ug.rst

```diff
@@ -0,0 +1,573 @@
+
+************
+ Objectives
+************
+
+The step-by-step instructions walk you through modifying a script for the purpose of performing the
+following functions:
+
+-  Report metrics
+-  Report checkpoints
+-  Perform a hyperparameter search
+-  Perform distributed training
+
+After completing the steps in this user guide, you will be able to do the following:
+
+-  Understand the minimum requirements for running an experiment
+-  Modify a script and an experiment configuration file
+-  Understand how to convert model code
+-  Use the Core API to train a model
+
+***************
+ Prerequisites
+***************
+
+**Required**
+
+-  A Determined cluster
+
+**Recommended**
+
+-  :doc:`Quickstart for Model Developers </quickstart-mdldev>`
+
+*****************************************************
+ Step 1: Get the Tutorial Files & Run the Experiment
+*****************************************************
+
+To run an experiment, you need, at minimum, a script and an experiment configuration (YAML) file.
+
+Create a new directory.
+
+Get the files from the :download:`core_api_pytorch_mnist.tgz </examples/core_api_pytorch_mnist.tgz>`
+download or from the `Github repository
+<https://github.com/determined-ai/determined/tree/master/examples/tutorials/core_api_pytorch_mnist>`_.
+
+.. note::
+
+   Throughout this user guide, we’ll show you how to make modifications to your script. After each
+   step, we’ll re-run our experiment using the appropriate script and its accompanying experiment
+   configuration (YAML) file.
+
+In this initial step, we’ll run our experiment using the ``model_def.py`` script and its
+accompanying ``const.yaml`` experiment configuration file.
+
+CD into the directory and run this command:
+
+.. code::
+
+   det e create const.yaml . -f
+
+Open the Determined WebUI by navigating to ``http://localhost:8080/``. Accept the default determined
+username, leave the password empty, and click **Sign In**.
+
+In the WebUI, select your experiment, and then navigate to the **Logs** tab.
+
+************************
+ Step 2: Report Metrics
+************************
+
+To report training and validation metrics to the Determined master, we’ll add a few lines of code to
+our script. More specifically, we'll create a ``core.Context`` object to allow interaction with the
+master. Then, we'll pass the ``core_context`` as an argument into ``main()``, ``train()``, and
+``test()`` and modify the function headers accordingly.
+
+To run our experiment, we'll use the ``model_def_metrics.py script`` and its accompanying
+``metrics.yaml`` experiment configuration file.
+
+Begin by importing Determined:
+
+.. code::
+
+   import determined as det
+
+.. image:: /assets/images/model-def-metrics-script.png
+
+Step 2.1: Modify the Main Loop
+==============================
+
+#. Modification of __main__ loop to include core_context
+
+Modify the __main__loop to include ``core_context``:
+
+.. note::
+
+   Refer to line 211 in ``model_def_metrics.py``
```

Comment:
Don't state a line number here, it would be wrong within 6 months and it would be basically impossible to detect.

If you look at my old user guide, I referred to this as "the `if __name__ == "__main__"` block" which python devs should have no trouble understanding.

Reply:
totally

Reply:
resolved